index.js

```diff
@@ -199,18 +211,27 @@ P.prototype._loadQuery = async function loadQuery(bbox, q) {
   self._recalc()
 }
 
-P.prototype._recalc = function() {
+P.prototype._recalc = function(fromScheduled) {
   var self = this
+  if (!fromScheduled && !self._recalcTimer) {
+    self._debug('something is calling _recalc() directly and nothing is scheduled')
```

Comment:
Here is a good point to add a `debugger` statement while developing. Nothing should call `_recalc()` directly, it should always be scheduled.